packages/synapse-react-client/src/components/EntityUpload/EntityUpload.tsx

```diff
@@ -0,0 +1,321 @@
+            <Typography variant={'smallText1'} my={2}>
+              {isLoading ? 'Loading...' : 'Preparing files for upload...'}
+            </Typography>
+          </>
+        )}
+        {!isPrecheckingUpload && !isLoading && (
+          <>
+            <img
+              src={
+                'https://s3.us-east-1.amazonaws.com/static.synapse.org/images/upload_illustration.svg'
+              }
+            />
+            {/* File input */}
+            <input
+              className="form-control"
+              type="file"
+              id="filesToUpload"
+              name="file upload"
+              style={{ display: 'none' }}
+              aria-hidden="true"
+              multiple={!isFileEntity}
+              ref={fileInputRef}
+              onChange={e => {
+                if (e.target.files != null) {
+                  uploadFileList(e.target.files)
+                }
+              }}
+            />
+            {/* Folder input */}
+            <input
+              className="form-control"
+              type="file"
+              id="folderToUpload"
+              name="folder upload"
+              style={{ display: 'none' }}
+              aria-hidden="true"
+              ref={folderInputRef}
+              onChange={e => {
+                if (e.target.files != null) {
+                  uploadFileList(e.target.files)
+                }
+              }}
+              // @ts-expect-error - webkitdirectory is not included in the React.InputHTMLAttributes type
+              webkitdirectory="true"
+            />
+
+            <Typography variant={'body1'} my={2}>
+              <Link onClick={handleClick}>Click to upload</Link> or drag and
+              drop
+            </Typography>
+            <Menu
+              anchorEl={anchorEl}
+              open={open}
+              onClose={handleClose}
+              anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+              transformOrigin={{ vertical: 'top', horizontal: 'left' }}
+              TransitionComponent={Fade}
+            >
+              <MenuItem
+                onClick={() => {
+                  handleClose()
+                  fileInputRef.current!.click()
+                }}
+              >
+                Files
+              </MenuItem>
+              <MenuItem
+                onClick={() => {
+                  handleClose()
+                  folderInputRef.current!.click()
+                }}
+              >
+                Folder
+              </MenuItem>
+            </Menu>
+
+            <Typography variant={'smallText1'}>
+              All uploaded files will be stored in
+              {uploadDestination?.storageLocationId ===
+                SYNAPSE_STORAGE_LOCATION_ID && ' Synapse storage'}
+              {uploadDestination?.storageLocationId !==
+                SYNAPSE_STORAGE_LOCATION_ID && (
+                <>
+                  :<br />
+                  {uploadDestination?.banner}
```

Comment:
Does this match the SWC implementation?  I would not think the banner text would always make sense after the "All uploaded files will be stored in" text.

Reply:
It does not match perfectly. Storage location objects do not have a name field. I think we have some code to display a storage location for a given file--maybe we can reuse that here and separate the banner.